packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart

```diff
@@ -121,11 +104,115 @@ class WidthPlatformBreakpoint extends Breakpoint {
+
+  /// Returns a [Breakpoint] with the given constraints for a large screen.
+  const Breakpoint.large({this.andUp = false, this.platform})
+      : beginWidth = 1200,
+        endWidth = 1600,
+        beginHeight = 900,
+        endHeight = null;
+
+  /// Returns a [Breakpoint] with the given constraints for an extraLarge screen.
+  const Breakpoint.extraLarge({this.andUp = false, this.platform})
+      : beginWidth = 1600,
+        endWidth = null,
+        beginHeight = 900,
+        endHeight = null;
+
+  /// A set of [TargetPlatform]s that the [Breakpoint] will be active on desktop.
+  static const Set<TargetPlatform> desktop = <TargetPlatform>{
+    TargetPlatform.linux,
+    TargetPlatform.macOS,
+    TargetPlatform.windows
+  };
+
+  /// A set of [TargetPlatform]s that the [Breakpoint] will be active on mobile.
+  static const Set<TargetPlatform> mobile = <TargetPlatform>{
+    TargetPlatform.android,
+    TargetPlatform.fuchsia,
+    TargetPlatform.iOS,
+  };
+
+  /// When set to true, it will include any size above the set width.
+  final bool andUp;
+
+  /// The beginning width dp value. If left null then the [Breakpoint] will have
+  /// no lower bound.
+  final double? beginWidth;
+
+  /// The end width dp value. If left null then the [Breakpoint] will have no
+  /// upper bound.
+  final double? endWidth;
+
+  /// The beginning height dp value. If left null then the [Breakpoint] will have
+  /// no lower bound.
+  final double? beginHeight;
+
+  /// The end height dp value. If left null then the [Breakpoint] will have no
+  /// upper bound.
+  final double? endHeight;
+
+  /// A Set of [TargetPlatform]s that the [Breakpoint] will be active on. If
+  /// left null then it will be active on all platforms.
+  final Set<TargetPlatform>? platform;
 
   /// A method that returns true based on conditions related to the context of
-  /// the screen such as MediaQuery.sizeOf(context).width.
-  bool isActive(BuildContext context);
+  /// the screen such as MediaQuery.sizeOf(context).width, MediaQuery.sizeOf(context).height
+  /// and MediaQuery.orientationOf(context).
+  bool isActive(BuildContext context) {
+    final TargetPlatform host = Theme.of(context).platform;
+    final bool isRightPlatform = platform?.contains(host) ?? true;
+
+    final double width = MediaQuery.sizeOf(context).width;
+    final double height = MediaQuery.sizeOf(context).height;
+    final Orientation orientation = MediaQuery.orientationOf(context);
+
+    final double lowerBoundWidth = beginWidth ?? double.negativeInfinity;
+    final double upperBoundWidth = endWidth ?? double.infinity;
+
+    final double lowerBoundHeight = beginHeight ?? double.negativeInfinity;
+    final double upperBoundHeight = endHeight ?? double.infinity;
+
+    final bool isWidthActive = andUp
+        ? width >= lowerBoundWidth
+        : width >= lowerBoundWidth && width < upperBoundWidth;
+
+    final bool isHeightActive = (orientation == Orientation.landscape &&
```

Comment:
This breaks desktops. Height is irrelevant for desktops, regardless of orientation, imo :(
@martijn00

Reply:
Fixed in https://github.com/flutter/packages/pull/7347